### PR TITLE
Add DateRangeBox Overview demo

### PR DESCRIPTION
**Pull request by alexanderPolosatov**

### CODE REVIEW

**Review by sergepilipchuk — DISMISSED** [review]

**Review by Krijovnick — DISMISSED** [review]

File: JSDemos/Demos/DateRangeBox/Overview/Angular/app/app.component.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+const msInDay = 1000 * 60 * 60 * 24;
+const now = new Date();
+const initialValue: [Date, Date] = [new Date(now.getTime() - msInDay * 3), new Date(now.getTime() + msInDay * 3)];
+
+function convertRangeToDays(value) {
```
Comment (ksercs) [inline JSDemos/Demos/DateRangeBox/Overview/Angular/app/app.component.ts]: Let's make this code more explicit
```suggestion
function convertRangeToDays([startDate, endDate]) {
```

File: JSDemos/Demos/DateRangeBox/Overview/Angular/app/app.component.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+import { DxDateRangeBoxModule } from 'devextreme-angular';
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+const msInDay = 1000 * 60 * 60 * 24;
+const now = new Date();
+const initialValue: [Date, Date] = [new Date(now.getTime() - msInDay * 3), new Date(now.getTime() + msInDay * 3)];
```
Comment (ksercs) [inline JSDemos/Demos/DateRangeBox/Overview/Angular/app/app.component.ts]:
```suggestion
const initialValue: [Date, Date] = [
    new Date(now.getTime() - msInDay * 3), 
    new Date(now.getTime() + msInDay * 3)
];
```

File: JSDemos/Demos/DateRangeBox/Overview/Angular/app/app.component.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  selectedDays: number = convertRangeToDays(initialValue);
+
+  constructor() {
+    const lastDay = new Date(now.getFullYear(), now.getMonth() + 1, 0).getDate();
+
+    this.min = new Date(now.setDate(1));
+    this.max = new Date(now.setDate(lastDay));
+  }
+
+  currentValueChanged({ value }) {
```
Comment (ksercs) [inline JSDemos/Demos/DateRangeBox/Overview/Angular/app/app.component.ts]: the same here, let's destructurize?

File: MVCDemos/Views/DateRangeBox/Overview.cshtml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+        if (startDate && endDate) {
+            daysCount = (endDate - startDate) / msInDay + 1;
+        }
+
+        $('#days-selected').text(daysCount);
+    }
+
+    $(function() {
+        showSelectedDays({ value: initialValue });
```
Comment (ksercs) [inline MVCDemos/Views/DateRangeBox/Overview.cshtml:92]: Do we really need wrap it in a function?

Comment (alexanderPolosatov) [reply MVCDemos/Views/DateRangeBox/Overview.cshtml:92]: it's equivalent of DOMContentLoaded event

**Review by Zedwag — APPROVED** [review]

**Review by marker-dao — APPROVED** [review]

**Pull request merged**